example/macos/Runner.xcodeproj/project.pbxproj

```diff
@@ -257,16 +257,19 @@
 			files = (
 			);
 			inputFileListPaths = (
+				"Flutter/ephemeral/FlutterInputs.xcfilelist"
 			);
 			inputPaths = (
+ 				"Flutter/ephemeral/tripwire"
 			);
 			outputFileListPaths = (
+				"Flutter/ephemeral/FlutterOutputs.xcfilelist"
 			);
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "\"$FLUTTER_ROOT\"/packages/flutter_tools/bin/macos_assemble.sh\n";
+			shellScript = "touch Flutter/ephemeral/tripwire\n\"$FLUTTER_ROOT\"/packages/flutter_tools/bin/macos_assemble.sh\n";
```

Comment:
Touching the tripwire should be last, not first, to ensure that it's newer than all the outputs even if things are created/copied by `assemble`

It would also help to put a comment on the line explaining why it's there and referencing an issue link.